lib/membrane/video_compositor/implementations/wgpu/native.ex

```diff
@@ -4,13 +4,30 @@ defmodule Membrane.VideoCompositor.Implementations.Wgpu.Native do
     otp_app: :membrane_video_compositor_plugin,
     crate: "membrane_videocompositor_wgpu"
 
+  alias Membrane.VideoCompositor.Implementations.OpenGL.Native.Rust.Position
   alias Membrane.VideoCompositor.Implementations.OpenGL.Native.Rust.RawVideo
 
-  @spec init(RawVideo.t(), RawVideo.t(), RawVideo.t()) :: {:ok, any()} | {:error, any}
-  def init(_first_video, _second_video, _out_video), do: error()
+  @type internal_state_t :: any()
+  @type error_t :: any()
+  @type id_t() :: non_neg_integer()
 
-  @spec join_frames(any(), binary(), binary()) :: {:ok, binary()} | {:error, any()}
-  def join_frames(_state, _upper, _lower), do: error()
+  @spec init(RawVideo.t()) :: {:ok, internal_state_t()} | {:error, error_t()}
+  def init(_out_video), do: error()
+
+  @spec join_frames(any(), [{id_t(), binary()}]) :: {:ok, binary()} | {:error, error_t()}
+  def join_frames(_state, _input_videos), do: error()
```

Comment:
Change spec of _state from `any()` to `internal_state_t`. I also sugest changing function name from `join_frames` to something like: `render_frame`, `compose_frames` or something similar.